File: ios/VizbeeVideo.m
//
//  VizbeeVideo.m
//  RNVizbeeSenderSdk
//

#import "VizbeeVideo.h"

/*
 
 // JS Onject for reference
 export default class VizbeeVideo {

     constructor() {

         this.guid = null;

         // metadata
         this.title = "";
         this.subtitle = "";
         this.imageUrl = "";
         this.isLive = false;

         // streamInfo
         this.streamUrl = null;
         this.tracks = {};
         this.startPositionInSeconds = 0;

         // custom
         this.customProperties = {};
     }
 }
 */

@implementation VizbeeVideo

-(instancetype) init:(NSDictionary*) vizbeeVideo {
    
    self = [super init];
    if (nil != self) {
        
        /** Globally Unique ID of the content */
        _guid = (NSString*) [vizbeeVideo objectForKey:@"guid"];

        /** Title of the video */
        _title = (NSString*) [vizbeeVideo objectForKey:@"title"];

        /** subTitle of the video */
        _subtitle = (NSString*) [vizbeeVideo objectForKey:@"subtitle"];

        /** Image URL */
        _imageUrl = (NSString*) [vizbeeVideo objectForKey:@"imageUrl"];

        /** True if the video is a live stream; false by default */
        NSNumber* isLiveObject = (NSNumber*) [vizbeeVideo objectForKey:@"isLive"];
        if (nil != isLiveObject) {
            _isLive = [isLiveObject boolValue];
        }

        /** streamUrl */
        _streamUrl = (NSString*) [vizbeeVideo objectForKey:@"streamUrl"];

        /** startPosition in seconds */
        NSNumber* positionObject = (NSNumber*) [vizbeeVideo objectForKey:@"startPositionInSeconds"];
        if (nil != positionObject) {
            _startPositionInSeconds = [positionObject doubleValue];
        }

        NSDictionary* customProperties = (NSDictionary*) [vizbeeVideo objectForKey:@"customProperties"];
        _customProperties = [NSDictionary dictionaryWithDictionary:customProperties];
    }
    
    return self;
}

-(NSDictionary*) toDictionary {
    
    NSDictionary* result = @{
        @"guid" : self.guid,
        @"title" : self.title,
        @"subtitle" : self.subtitle,
        @"imageUrl" : self.imageUrl,
        @"isLive" : [NSNumber numberWithBool:self.isLive],
        
        @"streamUrl" : self.streamUrl,
        @"startPositionInSeconds" : [NSNumber numberWithDouble:self.startPositionInSeconds],
        
        @"customProperties" : [NSDictionary dictionaryWithDictionary:self.customProperties]
    };
    return result;
}

@end

File: src/VizbeeVideo.js
/**
 * @class VizbeeVideo
 * @description Class representing a video in the Vizbee system
 */
export default class VizbeeVideo {
    /**
     * @constructor
     * @description Initializes a new instance of VizbeeVideo
     */
    constructor() {
        /**
         * @property {string|null} guid - Globally Unique Identifier for the video
         */
        this.guid = null;

        // metadata
        /**
         * @property {string} title - The title of the video
         */
        this.title = "";

        /**
         * @property {string} subtitle - The subtitle of the video
         */
        this.subtitle = "";

        /**
         * @property {string} imageUrl - URL of the video thumbnail or poster
         */
        this.imageUrl = "";

        /**
         * @property {boolean} isLive - Indicates if the video is a live stream
         */
        this.isLive = false;

        // streamInfo
        /**
         * @property {string|null} streamUrl - URL of the video stream
         */
        this.streamUrl = null;

        /**
         * @property {Object} tracks - Object containing track information
         */
        this.tracks = {};

        /**
         * @property {number} startPositionInSeconds - Start position of the video in seconds
         */
        this.startPositionInSeconds = 0;

        // custom 
        /**
         * @property {Object} customProperties - Custom properties associated with the video
         */
        this.customProperties = {};
    }
}